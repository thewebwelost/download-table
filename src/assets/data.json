[
  {
    "name": "smss.exe",
    "device": "Stark",
    "path": "\\Device\\HarddiskVolume2\\Windows\\System32\\smss.exe",
    "status": "scheduled"
  },
  {
    "name": "netsh.exe",
    "device": "Targaryen",
    "path": "\\Device\\HarddiskVolume2\\Windows\\System32\\netsh.exe",
    "status": "available"
  },
  {
    "name": "uxtheme.dll",
    "device": "Lannister",
    "path": "\\Device\\HarddiskVolume1\\Windows\\System32\\uxtheme.dll",
    "status": "available"
  },
  {
    "name": "cryptbase.dll",
    "device": "Martell",
    "path": "\\Device\\HarddiskVolume1\\Windows\\System32\\cryptbase.dll",
    "status": "scheduled"
  },
  {
    "name": "7za.exe",
    "device": "Baratheon",
    "path": "\\Device\\HarddiskVolume1\\temp\\7za.exe",
    "status": "scheduled"
  }
]
