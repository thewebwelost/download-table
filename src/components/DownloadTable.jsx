import React from 'react';
import SelectableTable from './SelectableTable';
import TableHeader from './TableHeader';

function DownloadTable({ data }) {
  return (
    <div>
      <TableHeader />
      <SelectableTable />
    </div>
  );
}

export default DownloadTable;
