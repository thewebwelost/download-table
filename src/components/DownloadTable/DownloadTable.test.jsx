import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import DownloadTable from '.';

import data from '../../assets/data.json';

afterEach(cleanup);

describe('<TableHeader />', () => {
  it('should render properly', () => {
    const tree = renderer.create(<DownloadTable data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render table rows correctly', () => {});
  it('should select all available rows', () => {});
  it('should deselect all available rows', () => {});
  it('should trigger download correctly', () => {});
});
