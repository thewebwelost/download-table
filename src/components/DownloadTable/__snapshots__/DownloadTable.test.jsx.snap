// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableHeader /> should render properly 1`] = `
<div>
  <header
    className="header"
  >
    <label
      className="selectAll"
      htmlFor="selectAllCheckbox"
    >
      <input
        checked={true}
        id="selectAllCheckbox"
        label="Select all rows"
        onChange={[Function]}
        type="checkbox"
      />
      Deselect all
    </label>
    <div>
      None selected
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        Download selected
      </button>
    </div>
  </header>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th />
        <th
          className="tableHead"
        >
          name
        </th>
        <th
          className="tableHead"
        >
          device
        </th>
        <th
          className="tableHead"
        >
          path
        </th>
        <th
          className="tableHead"
        >
          status
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="row
      unavailable 
      false"
        onClick={[Function]}
      >
        <td
          className="cell"
        >
          <input
            aria-label="select file smss.exe from Stark device"
            checked={false}
            disabled={true}
            onChange={[Function]}
            type="checkbox"
          />
        </td>
        <td
          className="cell"
        >
          smss.exe
        </td>
        <td
          className="cell"
        >
          Stark
        </td>
        <td
          className="cell"
        >
          \\Device\\HarddiskVolume2\\Windows\\System32\\smss.exe
        </td>
        <td
          className="cell"
        >
          scheduled
        </td>
      </tr>
      <tr
        className="row
      false 
      false"
        onClick={[Function]}
      >
        <td
          className="cell"
        >
          <input
            aria-label="select file netsh.exe from Targaryen device"
            checked={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
        </td>
        <td
          className="cell"
        >
          netsh.exe
        </td>
        <td
          className="cell"
        >
          Targaryen
        </td>
        <td
          className="cell"
        >
          \\Device\\HarddiskVolume2\\Windows\\System32\\netsh.exe
        </td>
        <td
          className="cell"
        >
          <div
            className="statusAvailable"
          />
          available
        </td>
      </tr>
      <tr
        className="row
      false 
      false"
        onClick={[Function]}
      >
        <td
          className="cell"
        >
          <input
            aria-label="select file uxtheme.dll from Lannister device"
            checked={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
        </td>
        <td
          className="cell"
        >
          uxtheme.dll
        </td>
        <td
          className="cell"
        >
          Lannister
        </td>
        <td
          className="cell"
        >
          \\Device\\HarddiskVolume1\\Windows\\System32\\uxtheme.dll
        </td>
        <td
          className="cell"
        >
          <div
            className="statusAvailable"
          />
          available
        </td>
      </tr>
      <tr
        className="row
      unavailable 
      false"
        onClick={[Function]}
      >
        <td
          className="cell"
        >
          <input
            aria-label="select file cryptbase.dll from Martell device"
            checked={false}
            disabled={true}
            onChange={[Function]}
            type="checkbox"
          />
        </td>
        <td
          className="cell"
        >
          cryptbase.dll
        </td>
        <td
          className="cell"
        >
          Martell
        </td>
        <td
          className="cell"
        >
          \\Device\\HarddiskVolume1\\Windows\\System32\\cryptbase.dll
        </td>
        <td
          className="cell"
        >
          scheduled
        </td>
      </tr>
      <tr
        className="row
      unavailable 
      false"
        onClick={[Function]}
      >
        <td
          className="cell"
        >
          <input
            aria-label="select file 7za.exe from Baratheon device"
            checked={false}
            disabled={true}
            onChange={[Function]}
            type="checkbox"
          />
        </td>
        <td
          className="cell"
        >
          7za.exe
        </td>
        <td
          className="cell"
        >
          Baratheon
        </td>
        <td
          className="cell"
        >
          \\Device\\HarddiskVolume1\\temp\\7za.exe
        </td>
        <td
          className="cell"
        >
          scheduled
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
